lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
```

Comment:
It's less a command than you're letting on, though. If this is `valid?`, there's no reason to check that `errors[:password].blank?` (would `empty?` make more sense here?)